### DISCUSSION TITLE
Declaring a relationship

### DISCUSSION THREAD

**Discussion by Arbiter-S** [sqlalchemy/sqlalchemy#10993] [Usage Questions] [answered]
This is part of document on how to work with ORM related objects.

```py
from sqlalchemy.orm import Mapped
from sqlalchemy.orm import relationship


class User(Base):
    __tablename__ = "user_account"

    # ... mapped_column() mappings

    addresses: Mapped[List["Address"]] = relationship(back_populates="user")


class Address(Base):
    __tablename__ = "address"

    # ... mapped_column() mappings

    user: Mapped["User"] = relationship(back_populates="addresses")
```

This is another part of document explaining relationship patterns.

```py
class Parent(Base):
    __tablename__ = "parent_table"

    id: Mapped[int] = mapped_column(primary_key=True)
    children: Mapped[List["Child"]] = relationship(back_populates="parent")


class Child(Base):
    __tablename__ = "child_table"

    id: Mapped[int] = mapped_column(primary_key=True)
    parent_id: Mapped[int] = mapped_column(ForeignKey("parent_table.id"))
    parent: Mapped["Parent"] = relationship(back_populates="children")
```

Since type annotations are not a necessity we can also make the first relationship like this.

```py
class User(Base):
    __tablename__ = "user_account"

    # ... mapped_column() mappings

    addresses = relationship("Address", back_populates="user")


class Address(Base):
    __tablename__ = "address"

    # ... mapped_column() mappings

    user_id = mapped_column(ForeignKey("user_account.id"))
    user = relationship("User", back_populates="addresses")
```

Do all of these achieve the same thing?

Also how does the difference between a bidirectional and a unidirectional relationship reflect in a database?
Wouldn't it be possible to go from a user instance to a address instances and other way around just by joining them based on their foreign key attribute?

**Comment by CaselIT** [comment, marked as answer]
Hi,

I'm not sure I completely understand the question but I'll give it a shot

> Do all of these achieve the same thing?

yes, but without annotation you may need to set `uselist` in some cases

> Also how does the difference between a bidirectional and a unidirectional relationship reflect in a database?

if I understand correctly what you mean, the db structure is the same, relationship are just a python construct to improve usability in common cases

> Wouldn't it be possible to go from a user instance to a address instances and other way around just by joining them based on their foreign key attribute?

sure

**Reply by Arbiter-S** [reply]
Appreciate the response
